razorpay-payments.php

```diff
@@ -329,13 +335,94 @@ protected function createRazorpayOrderId($orderId, $sessionKey)
             $api = $this->getRazorpayApiInstance();
 
             $data = $this->getOrderCreationData($orderId);
+
+            // Validate the order
+            try
+            {
+                $this->validateOrderCreateData($data);
+            }
+            catch(Errors\InvalidCurrencyError $e)
+            {
+                // If currency is not INR, check if Woocommerce Currency Switcher plugin exists
+                if (class_exists('WOOCS'))
+                {
+                    $data = $this->convertCurrencyWoocs($data);
+                }
+                // If the plugin does not exist throw an exception
+                else
+                {
+                    throw new Errors\BadRequestError(
+                        Errors\ErrorCode::WOOCS_MISSING_ERROR_MESSAGE,
+                        Errors\ErrorCode::WOOCS_MISSING_ERROR_CODE,
+                        400
+                    );
+                }
+            }
+
             $razorpay_order = $api->order->create($data);
 
             $razorpayOrderId = $razorpay_order['id'];
 
             $woocommerce->session->set($sessionKey, $razorpayOrderId);
 
             return $razorpayOrderId;
+
+        }
+
+        /**
+         * Validate the order creation data before calling the order create API
+         *
+         * @param  Array $data The data passed to create order API
+         *
+         * @return void
+         */
+        protected function validateOrderCreateData($data): Array
```

Comment:
The flow here seems to be using exceptions pretty loosely. Instead of:

1. Setup data
2. Validate (if, check, throw)
3. Catch-> Fix
4. Proceed

Why not just check the conditions upfront:

```
if(INR) ->
else if ($installed and not INR)
else if(not INR)
```